telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/statistics/service/StatisticsService.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.bumblebee.command.statistics.service
+
+import com.github.bumblebee.command.statistics.dao.StatisticsRepository
+import com.github.bumblebee.command.statistics.entity.Statistic
+import org.springframework.stereotype.Service
+import java.lang.StringBuilder
+import java.time.LocalDate
+
+@Service
+class StatisticsService(private val repository: StatisticsRepository) {
+
+    private val existingStatistics: MutableMap<Long, List<Statistic>> = fillStatisticsMap()
+
+    private fun getAllUserStatisticsForCurrentDayAndChat(chatId: Long): List<Statistic>? {
+        return existingStatistics[chatId]
+                ?.filter { it.postedDate == LocalDate.now() }
+    }
+
+    private fun fillStatisticsMap(): MutableMap<Long, List<Statistic>> {
+        return getAllUserStatisticsForCurrentDay().groupBy { it.chatId }.toMutableMap()
+    }
+
+    fun getChatsWithStatistic(): Set<Long> {
+        return existingStatistics.keys
+    }
+
+    private fun getAllUserStatisticsForCurrentDay(): List<Statistic> {
+        return repository.findStatisticByPostedDate(LocalDate.now()).orEmpty()
+    }
+
+    private fun getSortedStatisticsByMessagesCountInChat(chatId: Long): List<Statistic>? {
+        return getAllUserStatisticsForCurrentDayAndChat(chatId)?.sortedBy { statistic -> statistic.messageCount }
+    }
+
+    private fun getUserStatisticForCurrentDayInChat(userId: Long, chatId: Long): Statistic? {
+        //return repository.findStatisticByUserIdAndPostedDateAndChatId(userId, LocalDate.now(), chatId)
+        val allStatForChat = getSortedStatisticsByMessagesCountInChat(chatId)
+        if(allStatForChat != null) {
```

Comment:
You could use conditional calls here: getSortedStatisticsByMessagesCountInChat(chatId)?.find...